src/components/PhoneNumberInput/index.js

```diff
@@ -18,25 +18,47 @@ const FlagComponent = ({ countryCode, flagsPath }) => (
   />
 );
 
-const PhoneNumberInput = ({ translate, clearErrors, actions = {}, sms = {}, smsNumberCountryCode}) => {
+class PhoneNumberInput extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      validPredefinedNumber: false,
+    }
+  }
+
+  componentDidMount() {
+    const smsProps = this.props.sms
+    if (smsProps && smsProps.number) {
+      const validPredefinedNumber = isValidPhoneNumber(smsProps.number)
+      this.setState({validPredefinedNumber})
```

Comment:
why is state necessary if the number is on the store?

Reply:
It's not necessary